### DISCUSSION TITLE
Undefined variable error

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [symfony/symfony#46786] [Q&A] [answered]
Hello everyone !
I m working on projet and i have method:
![image](https://user-images.githubusercontent.com/101296742/175895679-3d6bec27-9c31-4aa0-b9cb-1f440b033f9e.png)
And i call it in this route : 
![image](https://user-images.githubusercontent.com/101296742/175895851-3f15212a-cbd2-4034-b627-10a378a50daa.png)
But i keep getting this error 
![image](https://user-images.githubusercontent.com/101296742/175895932-32cd6abb-cee4-4420-ab57-2d769b02d366.png)
Thank you !

**Comment by stof** [comment, marked as answer]
Well, I see 2 cases that can end up not defining that variable:

- when the for loop iterates 0 times
- when the `if` inside that loop is falsy for all iterations.

**Reply by Elmatbaisaad** [reply]
thank you i will try figure out in which i am
